streams/src/main/java/org/apache/kafka/streams/errors/SerializationExceptionHandler.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.errors;
+
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.Configurable;
+import org.apache.kafka.common.serialization.Serializer;
+
+/**
+ * Interface that specifies how an exception while attempting to serialize the
+ * the record should be handled
+ */
+public interface SerializationExceptionHandler extends Configurable {
+
+    /**
+     * Inspecting a record that we attempted to serialize, and the exception that resulted from
+     * attempting to serialize it and determine whether or not continue processing.
+     *
+     * @param record        the record that failed to serialize
+     * @param isKey         denotes whether the error occurred while serializing the record key or value.
+     *                      <code>true</code> for record key; <code>false</code> for record value
+     * @param serializer    serializer which serialized the record key / value
+     * @param exception     the actual exception
+     */
+    default SerializationExceptionHandlerResponse handle(final ProducerRecord record,
+                                                         final boolean isKey,
+                                                         final Serializer serializer,
+                                                         final Exception exception) {
+        return SerializationExceptionHandlerResponse.FAIL;
```

Comment:
Why do we need to have this default here? Is it just to be symmetric with DefaultProductionExceptionHandler or are there any reasons behind it?

Reply:
Yes, I added it for symmetric purpose.

Reply:
Removed this method.